### ISSUE TITLE
Try to determine type of variable

### ISSUE THREAD

**Issue by dustinspecker** [dustinspecker/eslint-plugin-no-use-extend-native#4]
As mentioned in https://github.com/sindresorhus/module-requests/issues/25.

Investigate possibility of looking at variable declarations and assignments to understand the type.

This way things such as:

``` javascript
var test = []; // some-library.js modified Array's prototype
test.customFunction(); // => ESLint should report as an issue
```

**Comment by sindresorhus** [comment]
Sounds like something that should be built into ESLint and available to plugins. I know they're working on path analysis, which sounds similar to this. You might want to join that discussion: https://github.com/eslint/eslint/issues/3373

**Comment by sindresorhus** [comment]
ESLint path analysis pull request in case you have any feedback: https://github.com/eslint/eslint/pull/3559

**Comment by dustinspecker** [comment]
Almost 10 years later, I'm going to close this for now. 😃
